cmd/check-gh-automation/main.go

```diff
@@ -187,6 +190,30 @@ func checkRepos(repos []string, bots []string, appName string, ignore sets.Set[s
 			repoLogger.Info("no bots provided to check")
 		}
 
+		orgConfig := prowAgent.Config().BranchProtection.GetOrg(org)
+		var branchProtected bool
+		if orgConfig != nil && orgConfig.Repos != nil {
+			_, branchProtected = orgConfig.Repos[repo]
+		}
+
+		// If branch protection is configured, verify admin access for 'openshift-merge-robot' and bots in 'adminBots'
+		if branchProtected {
+			logger.Infof("Branch protection is enabled for %s/%s", org, repo)
+			for _, bot := range append(adminBots, "openshift-merge-robot") {
+				hasAdminAccess, err := client.HasPermission(org, repo, bot, "admin")
+				if err != nil {
+					logger.Errorf("Error checking admin access for bot %s in %s/%s: %v", bot, org, repo, err)
+					return nil, fmt.Errorf("error checking admin access for bot %s in %s/%s: %w", bot, org, repo, err)
+				}
+				if !hasAdminAccess {
+					logger.Warnf("Bot %s does not have admin access in %s/%s with branch protection enabled", bot, org, repo)
```

Comment:
I guess this log would be better at the `error` level, it will ultimately cause the check to fail and will be more clear to the user what the problem is